### DISCUSSION TITLE
Discuss common MEs here

### DISCUSSION THREAD

**Discussion by Tempo-anon** [Despair-Games/poketernity#132] [Balance]
### Berries abound ⭐ ⭐ ⭐ 
Flavor: A strong Pokemon guarding berries makes sense from a lore point of view and also blends together some gameplay elements together. It definitely works
Gameplay: It feels a bit underwhelming compared to other MEs, but I think it works fine for being something a little more unique than a randomly generated opponent. I just think that berries as a whole are pretty underwhelming rewards even if they get a bit more utility from other MEs
Code: Nothing too egregious here, and an okay enough framework for extending into different opponents, different rewards, and different options to bypass the battle

### Department Store Sale ⭐ ⭐ ⭐ ⭐ ⭐ 
Flavor: a giant shopping store is a recurring location in all the mainline games so this fits right in.
Gameplay: An option for specifically choosing a reward category (TM, vitamin, pokeball, x item) is a fantastic way of giving players more choices and rewarding good decisions
Code: Seems pretty simple and really scalable for other events with different reward options

### FIeld Trip ⭐ ⭐ ⭐ ⭐ 
Flavor: Pokemon schools are a common occurrence and this is a cute interaction
Gameplay: Not every ME needs to be some flashy banger. This is a very simple introduction to MEs as well as beginner players while also being a check for button mashers to make sure they aren't asleep. The rewards are simple but should be satisfying. Should be kept as an early game (<50) ME though
Code: Also not too bad to be a springboard for building more quiz like scenarios

### Fiery Fallout ⭐ ⭐ 
Flavor: Having to endure a firestorm is a neat idea but I wonder if having it always be Volcarona 
Gameplay: I have a few gripes with this ME. The combat option always being Volcarona instead of other possible fire Pokemon makes this less unique than the other MEs that feature tough fights. The hunkering down option seems too potentially punishing (changing a random ability to heatproof), and the final option has a pretty lenient requirement (fire type or fire resistance ability). I think the balancing on this definitely needs work
Code: Actually pretty clean and scalable

### Lost at Sea :star: :star: :star: :star:
Flavor: I think every player knows this feeling so capturing it in an ME is a fantastic idea
Gameplay: I quite like this. It's a simple team check that results in some HP penalty if the player doesn't meet it, but the requirement is generous enough (having a Pokemon that can learn surf/fly)
Code: Pretty clean and scalable

### Part Timer :star: :star: :star: :star:
Flavor: A way to have a Pokemon doing work in exchange for money fits very well into the worldbuilding of Pokemon
Gameplay: It's simple but I like it. A quick team stat check for stats in exchange for money (which has improved value due to MEs) is pretty good in lieu of a normal encounter. The "penalty" of reducing PP is also pretty good since it touches on a mechanic that is not too commonly interacted with in the game.
Code: Not too bad but reading a lot of ME code makes me think that a lot of stuff can be moved to helper functions, especially when a choice is made

### Teleporting Hijinks :star: :star:
Flavor: A way to jump to a random biome is something players really want and is quite a fun gameplay mechanic that could use further exploration. There is also an option to bypass the money requirement which is pretty cute
Gameplay: I love the idea of a random biome teleport but this teleport only has space, fairy cave, lab, island, wasteland, and dojo which is sort of ?_? to me. Additionally, I don't know if forcing a boss encounter is really necessary. I feel like this is one of those MEs that has a lot of potential but the way it's currently implemented could use some tweaking
Code: Some minor copying/pasting overall okay

### GTS :star: :star: :star: :star:
Flavor: The GTS and its equivalents are some of the most fun in single player Pokemon games because it gets to both simulate the experience of a trade as well as getting a random Pokemon
Gameplay: Getting a random Pokemon or a choice of a new Pokemon is very fun and exciting. Also providing metaprogression through unlocks makes this even more desirable. Egg move slots should be part of the metaprogression unlocks though.
Code: Helper functions would help this a lot but this could and should be expanded for more trading events

### Mysterious Chest :star: :star: :star:
Flavor: A "mimic" in the Pokerogue sense, I think this works
Gameplay: Perhaps the very first ME created, it's a simple risk vs reward decision. Would prefer to not have the same boss with hardcoded moveset each time but I think this ME is fine.
Code: Seeing some more duplicated code

### Vitamin dealer :star: :star: :star: :star:
Flavor: I think we all know what the shady vitamin dealer is a stand-in for but I think it works okay with enough plausible deniability
Gameplay: I think this is a good ME in that it is one of the MEs that really makes money a more valuable resource during a run. Getting permanent boosts is always a great feeling and being able to convert your money to extra stats is great. Being able to randomize a nature is also interesting but it doesn't update the dex which may cause some confusion to players. Things that give the player new abilities/natures/moves should be consistent in whether or not they actually provide permanent metaprogression
Code: Not too terrible, can potentially be used for other permanent stat gain events

Uncommon Breed :star: :star: :star:
Flavor: It's another variation on the "wild encounter but stronger" but the twist fails to land as well because it's not unlockable
Gameplay: Not unlocking the egg moves permanently is criminal. One of the most important things in the game should be allowing more metaprogression unlocks so arbitrarily not allowing the permanent unlock is weird. If the move unlocks were permanent this could have easily been a 5 star event 
Code: Since this shares the base idea with berries abound, it makes me think there should be something in ME utils for better scalability of "wild encounter but with a twist"

**Comment by CodeTappert** [comment]
My thoughts: (I dont care about the code in judging if we should keep or change it)

Berries around: It may be good flavour wise but the gameplay affect is not good. You mostly just pick fight to get all your stuff back. Otherwise this would be a net minus ME. Maybe remove the steal part.

Department Store Sale: Keep as is. Its a funny idea and the player can choose what he wants. Not sure if currently possible but there should be a "back and choose different store" option. So a player can look at all of them and then deicde.

Field Trip: Keep as is

Fiery Fallout: Make it different pokemon species. Either random fire pokemon or from a given list. 

Lost at Sea: I think the wording in the ME itself is little off. (Or maybe thats fixed by now but it was when translating it). Keep otherwise as is

Part Timer: Also i think the wording is weird. Flavour and gameplay wise fine

Teleporting Hijinks: Make every biome (no end biome) possible as teleport. 

GTS: Feels weird.Since they seem not really random but a set list? 

Mysterious Chest: Its fine. 

Vitamin Dealer: Keep as is

UNcommon Breed: Should unlock whatever you catch/get.

**Comment by gerolau** [comment]
im boutta yap

Berries abound:
Fine ME, teaches the player the benefits of having a Pokemon have stealing moves on it. Berries are an underwhelming reward but for a common ME that's okay. Might be more of use to new players who don't always utilize Pick Up

Department Store Sale:
Fine, however I truly have never seen a reason to pick anything besides the TM shop, and then most the time the TMs end up not being the most helpful/not being rerollable due to how the code works. I don't think this is innately bad, but until there is a more proper and reliable TM acquiring method it just feels like a luck based roll of 5 TMs.

Field Trip:
I see the qualms of saying its too free/no one will take the x items but mentioning capping it before floor 50/ considering it a boost to new players or fresh start runs it's likely fine as is. maybe a second condition could be added to her question, like a specific type, and if you show a move of that type while also matching the physical/special/status question the rewards can be ultra tier or even mints.

Fiery Fallout:
This is intended to be a negative ME, but even with that context it has flaws

- volcano already sucks to get to and be in, so having a harmful ME here give you even less reason to go
- to an unexperienced player the volcaronas may be too over juiced, variety in what you encounter is nice
- further more, by being volcaronas, rock slide trivializes the fight.
- changing to heatproof isnt bad but the randomness is; if i could pick the pokemon to get it to overwrite a null ability or give it to a mon with a fire weakness i may honestly take the consequences over fighting the volcaronas

im meh on the resistant abilities list but most seem fitting. maybe slap magic guard in there too

Lost at Sea 
fully agree here, no notes. a very good negative ME

Part Timer
I quite like this one as is, this said, the charming moves list is so horrifically bad that the 3rd option will likely always be extremely niche.
For reference

- Alluring Voice (most likely to actually have)
- Aromatic Mist
- Attract
- Captivate
- Charm (if you have some fucked up prankster set maybe)
- Dragon Cheer
- Flatter
- Sweet Scent

Teleporting Hijinks
The original intent of the locked few biomes was it was supposed to be a way to reach rarer biomes, but since this is less of an issue in current versions of the game it just feels like arbitrary limits. It could probably have a very high chance to hit those rare biomes in specific, or maybe it could offer you a choice of 3 random biomes and you can pick

while i dont hate the boss fight (again another weird asdar balance choice with the intent of helping people find rare bosses) i can see some people actively not using the teleporter if they dont think they can take it, which im not sure if we want.

GTS
I think the egg move on the traded mon is fine, partially because its a compensation for not rolling the other two benefits of the option, I myself never really minded not having the egg move, but if its for the sake of adding more natural metaprogression that could be very nice, especially since the other option is at its best when passives are unlocked. this said, maybe rare moves should be banned or weighted lower here. While egg moves are immensely strong, it is still a luck check on if you get useful ones or niche ones. While I quite liked my ursaluna with fake out, its no free precipice blades. 

Honestly, i had no idea if you take a pokemon with the passive unlocked it gives the passive, i mustve always skimmed it too quick. this could have questionable results but again its a luck check. 

If i get this late game i usually just sack an eviolite on the item trade and move on, which is nice so you dont have to be forced to trade a mon

Mysterious Chest
I know the whole point is it's a play on a mimic, but for a common ME i think an instakill is extremely harsh. I feel like we could maybe just make the battle lead with Curse applied to the front pokemon, to still cause the player to likely need to switch off their carry without potentially sending a baby profile back to the starter screen. I know for experienced players its very easy to play around, so maybe im being a bit too difficulty wary.

Vitamin dealer 
I agree unlocking the nature should be fine considering the mints already do that, plus it'd actually give more benefit to rolling the cheap deal. Overall fine, vitamins are always appreciated.

Uncommon Breed
Unlocking them is fine yeah. These MEs that start with stat boosts are all completely trivialized by opportunist, idk if thats a concern or not.

**Comment by innerthunder** [comment]
Rating in terms of gameplay only...

### Berries Abound 5/10

- Works fine as a simple ME in the early game, but encountering it later on is underwhelming.
- The rewards aren't compatible with the restrictions on held item slots that we're considering.
- The race option is too easy to consistently succeed at, assuming you know the opposing Pokemon and have some knowledge/info on their base stats. There's very little randomness or risk involved.

### Department Store Sale 7/10

- A simple ME with decent rewards overall
- The TM and Vitamin Shops are a lot better than the other shops in most situations

### Field Trip 4/10

- The encounter is cute, but the rewards are really underwhelming
- Rarer Candy being the best reward in a lot of situations takes away from the impact of the player's move selection

### Fiery Fallout 7/10

- I like its flavor and uniqueness as a Volcano exclusive
- The Volcaronas are intended to be overpowered, but the stat boosts and free moves might be too much
- Having "Hunker Down" change a random Pokemon's ability is too harsh of a penalty

### Fight or Flight 8/10

- Straightforward, the item offered is usually good
- Could bring up the item's tier in earlier waves to make it more enticing

### Lost at Sea 9/10

- Great flavor and a good approach to negative effects in an ME

### Part-Timer 6/10

- Cool concept of putting a Pokemon to work with pros and cons
- In practice, setting the Pokemon's PP to 2 isn't much of a downside
- The Charming moves don't see much use overall. Maybe the option could be more rewarding?

### Teleporting Hijinks 4/10

- The biome restrictions are kind of arbitrary now that rarer biomes are generally easier to access
- The amped up boss encounter after teleporting doesn't seem necessary

### The GTS 6/10

- Cool in concept and flavor
- Wonder Trading for low shiny/HA odds and arbitrarily gated metaprogression isn't worth selecting over the trade offers
- I usually prefer to trade held items in this ME since I usually have a filler Ultra-tier item at that point and most Rogue-tier items are a net positive.

### The Mysterious Chest 7/10

- It's fine as a simple risk/reward
- It might be worth increasing the number of reward options for the ultra/epic-tier rewards

### The Vitamin Dealer 8/10

- Good payoff for saving money
- Why is the random nature locked from dex progression?

### Uncommon Breed 5/10

- Simple hard encounter + payoff for hoarding berries
- Why is the random egg move locked from dex progression?
- Again, the set of Charming moves is underwhelming

**Comment by AsdarDevelops** [comment]
Hello all. My two cents on this:

I think the worst two offenders are Mysterious Chest and Fiery Fallout, which both deviated from the design principle that a Common ME should be simple, clean (at least outwardly facing) and very repeatable, yet with somewhat varied outcomes.

## Teleporting Hijinks 2/10 (the very forced ME)

Was really hard to put this idea into gameplay as it seems good on paper but the execution failed to respect the principles that we went by when designing. Ideally this would be a "0 floor cost" ME. The ME also meant to give access to very hard-to-reach biomes, which are not so unreachable anymore.

If the fights feel shoehorned in it's because they are, they were added to not negatively impact the player who is already missing a floor.
The asset is really nice. My solution for this would be to let the player choose any biome they want to go to, and move the ME up in rarity.

## Mysterious Chest 3/10

The OG indeed.

Mysterious Chest didn't have a fixed battle and I don't think adding one was the correct choice. There was a Goldengho-themed ME concept which got pushed to Ultra and then discarded for this reason. I agree with @gerolau that a curse rather than a straight up KO would be more interesting. The Gimmighoul fight also sort of gives the impression that, the other times you got your pokemon KOd, it was the teeny Gimmighoul doing it.

This was very much designed with the idea of disincentivizing one-pokemon parties, which is a game design principle I stand by but others might disagree with. Worth a more profound discussion imo.

## Fiery Fallout 5/10

Fiery Fallout was initially considered as rare ME to avoid repeated fixed battles. This ME has a Pokemon that is very dear to the dev that made it, one of his favourites. Even if it's Volcano locked, I dislike the fact that the battle is fixed. I would suggest pushing it up in rarity and adjusting the rewards accordingly to something more powerful. The ME was never intended to be a "negative" for the player, I feel that was the justification being thrown around after receiving complaints about it.

Animations are insanely cool and definitely not fitting for a common.

## The rest

I think the rest are all fine, mostly short and sweet, doing different checks on team diversity. I'd have to revisit the list for charming moves that @innerthunder mentioned but I think there weren't that many and that's why the list is somewhat short and weak.

## My references

My reference point for Common MEs would be "Fight or Flight" and "Lost at Sea". The code for Lost at Sea was made by @flx-sta in such a way that it could accommodate other designs (boringly named "Lost at Forest, Swamp, etc..." that were all meant to be Common so that the pool of Commons was a neutral occurrence, more or less, rather than a mostly positive.

## Regarding consistency in metaprogression.

Many things like egg/eggmoves/natures/higher IVs are locked (not given to dex) in Common encounters because it was imposed on the design by the balance division at the time. Dex unlocks are mostly reserved to ultra +, save some exceptions.
